test/performance/range/views/view_translate_2D_1D_benchmark.cpp

```diff
@@ -0,0 +1,227 @@
+void sequential_read_impl(benchmark::State & state, rng_t && rng)
+{
+    for (auto _ : state)
+    {
+        for (auto && outer : rng)
+        {
+            for (aa27 inner : outer)
+            {
+                benchmark::DoNotOptimize(inner.to_rank());
+            }
+        }
+    }
+}
+
+template <typename tag_t>
+void sequential_read(benchmark::State & state)
+{
+    std::vector<std::vector<dna4>> c;
+    c.resize(1000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+        c[i] = test::generate_sequence<seqan3::dna4>(100, 0, 0);
+
+    if constexpr (std::is_same_v<tag_t, baseline_tag>)
+    {
+        std::vector<std::vector<aa27>> cpy = c | views::translate_join | views::to<std::vector<std::vector<aa27>>>;
+        sequential_read_impl(state, cpy);
+    }
+    else if constexpr (std::is_same_v<tag_t, translate_tag>)
+    {
+        auto v = c | views::translate | views::join;
+        sequential_read_impl(state, v);
+    }
+    else
+    {
+        auto v = c | views::translate_join;
+        sequential_read_impl(state, v);
+    }
+}
+
+BENCHMARK_TEMPLATE(sequential_read, baseline_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_join_tag);
+
+// ============================================================================
+//  random_access
+// ============================================================================
+
+template <typename rng_t>
+void random_access_impl(benchmark::State & state, rng_t && rng,
+                        std::vector<size_t> const & access_positions_outer,
+                        std::vector<std::vector<size_t> > const & access_positions_inner)
+{
+    for (auto _ : state)
+    {
+        for (size_t i = 0; i < access_positions_outer.size(); ++i)
+        {
+            for (size_t j = 0; j < access_positions_inner[i].size(); ++j)
+            {
+                benchmark::DoNotOptimize(rng[i][j].to_rank());
+            }
+        }
+    }
+}
+
+template <typename tag_t>
+void random_access(benchmark::State & state)
+{
+    std::vector<std::vector<dna4>> c;
+    c.resize(1000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+        c[i] = test::generate_sequence<seqan3::dna4>(200, 0, 0);
+
+    std::vector<size_t> access_positions_outer;
+    access_positions_outer.resize(200);
+    std::mt19937 gen(42);
+    std::uniform_int_distribution<size_t> dis_position_outer(0, 1000 - 1);
+
+    for (size_t i = 0; i < 200; ++i)
+        access_positions_outer[i] = dis_position_outer(gen);
+
+    std::vector<std::vector<size_t> > access_positions_inner;
+    access_positions_inner.resize(200);
+    for (size_t i = 0; i < 200; ++i)
+        access_positions_inner[i].resize(20);
+    std::uniform_int_distribution<size_t> dis_position_inner(0, 50 - 1);
+
+    for (size_t i = 0; i < 200; ++i)
+    {
+        for (size_t j = 0; j < 20; ++j)
+        {
+            access_positions_inner[i][j] = dis_position_inner(gen);
+        }
+    }
```

Comment:
Ah, ok, got it. Well it's a little confusing with so many different value spans...

Is it not possible to e.h. use 150 sequences of length 150 and then use 150 random outer positions and 150 random inner positions? (You can use the same random inner positions every time)

I mean, it's not important, but it might make it easier to read and tune. (Just have one `constexpr size_t count = ` at the top of the function that can easily be changed).

Reply:
Yes, I thought about it initially but wasn't sure whether this was what the random access should depict. I can change this.